batik-gvt/src/main/java/org/apache/batik/gvt/font/AWTGlyphGeometryCache.java

```diff
@@ -39,160 +40,55 @@ public class AWTGlyphGeometryCache {
     protected static final int INITIAL_CAPACITY = 71;
 
     /**
-     * The underlying array
+     * The underlying map
      */
-    protected Entry[] table;
-
-    /**
-     * The number of entries
-     */
-    protected int count;
-
-    /**
-     * The reference queue.
-     */
-    protected ReferenceQueue referenceQueue = new ReferenceQueue();
+    ConcurrentHashMap<Character, SoftReference<Value>> cache;
 
     /**
      * Creates a new AWTGlyphGeometryCache.
      */
     public AWTGlyphGeometryCache() {
-        table = new Entry[INITIAL_CAPACITY];
+        this(INITIAL_CAPACITY);
     }
 
     /**
      * Creates a new AWTGlyphGeometryCache.
      * @param c The inital capacity.
      */
     public AWTGlyphGeometryCache(int c) {
-        table = new Entry[c];
+        cache = new ConcurrentHashMap<>(c);
     }
 
     /**
      * Returns the size of this table.
      */
     public int size() {
-        return count;
+        return cache.size();
     }
 
     /**
      * Gets the value of a variable
      * @return the value or null
      */
     public Value get(char c) {
-        int hash  = hashCode(c) & 0x7FFFFFFF;
-        int index = hash % table.length;
-
-        for (Entry e = table[index]; e != null; e = e.next) {
-            if ((e.hash == hash) && e.match(c)) {
-                return (Value)e.get();
-            }
-        }
-        return null;
+        return Optional.ofNullable(cache.get(c)).map(SoftReference::get).orElse(null);
```

Comment:
This creates two temporary Optionals. I suggest adding a helper method:

```
private static <T> T getIfSet(SoftReference<T> ref) {
    return ref == null ? null : ref.get();
}
```
and then

```
public Value get(char c) {
        return getIfSet(cache.get(c));
}
```